src/key.rs

```diff
@@ -1216,6 +1215,26 @@ impl FromStr for Key {
     }
 }
 
+#[cfg(feature = "serde")]
+impl serde::Serialize for Key {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&format!("{}", self))
+    }
+}
+
+#[cfg(feature = "serde")]
+impl<'de> serde::Deserialize<'de> for Key {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where D: serde::Deserializer<'de>
+    {
+        let s = String::deserialize(deserializer)?;
+        FromStr::from_str(&s).map_err(serde::de::Error::custom)
```

Comment:
It's a bit more code but consider deserializing using a Visitor with a visit_str method to avoid copying into a String.